hasty/bucket.py

```diff
@@ -0,0 +1,96 @@
+from collections import OrderedDict
+from typing import Union
+
+from constants import BucketProviders
+from .hasty_object import HastyObject
+
+class Credentials:
```

Comment:
This should be typing.Protocol with the relevant methods. Something equivalent to an interface + method signatures in golang.

I guess the whole structure also can be defined as a dataclass with same properties and an asdict method  for getting the credentials.

But none of these are Must-Have's.